### PR TITLE
chore: Rename the Terraform workflow file steps

### PR DESCRIPTION
**Pull request by gatici**
- Refactoring the steps of workflow file for Terraform checks.
- Adding LICENSE file
-  Adding license headers for the workflow files

### CODE REVIEW

File: .github/workflows/terraform.yaml [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
 
-      - name: HashiCorp - Setup Terraform
+      - name: Setup Terraform
         uses: hashicorp/setup-terraform@v3
 
-      - name: Terraform Format
+      - name: Format Terraform Files
         id: fmt
         run: terraform fmt -recursive -check
         continue-on-error: true
```
Comment (gruyaume) [inline .github/workflows/terraform.yaml:20]: Why do we have this stage at all? This is used to reformat the terraform files but we won't do anything with the output

File: .github/workflows/terraform.yaml [hunk 2]
```diff
@@ -1,28 +1,48 @@
...
+          set +e 
+
+          echo "CHECK_STATUS=0" >> $GITHUB_ENV
+
+          for changed_file in ${{ steps.pr_files.outputs.added_modified }}; do 
+            echo "Checking terraform fmt on ${changed_file}..."
+
+            if [[ $changed_file == *.tf ]]; then 
+              terraform fmt -check $changed_file
+              FMT_STATUS=$(echo $?)
```
Comment (markbeierl) [inline .github/workflows/terraform.yaml:38]: you can just use `FMT_STATUS=$?`, or even right to the next line like
```
terraform fmt -check $changed_file
if [[ $? -ne 0 ]]; then
...
```

File: .github/workflows/terraform.yaml (hunk 1 above)
Comment (gatici) [reply .github/workflows/terraform.yaml]: `terraform fmt` output is processed.

File: .github/workflows/terraform.yaml (hunk 2 above)
Comment (gatici) [reply .github/workflows/terraform.yaml]: Done

File: .github/workflows/terraform.yaml [hunk 3]
```diff
@@ -1,28 +1,46 @@
...
+          echo "CHECK_STATUS=0" >> "$GITHUB_ENV"
+          terraform fmt -recursive -check
+          FMT_STATUS="$?"
+          if [[ "$FMT_STATUS" -ne 0 ]]; then
+            echo "❌ terraform fmt failed" >> "$GITHUB_STEP_SUMMARY"
+            echo "CHECK_STATUS=1" >> "$GITHUB_ENV"
+          fi
+      - name: Process check
+        if: always()
+        run: exit "$CHECK_STATUS"
```
Comment (Gmerold) [inline .github/workflows/terraform.yaml:46]: Does this work? Shouldn't it depend on `Validate terraform fmt`?

**Review by Gmerold — CHANGES_REQUESTED** [review]

File: LICENSE [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
[the remark this answers is not in the document]
Comment (gatici) [reply LICENSE:189]: This is updated.

File: .github/workflows/terraform.yaml [hunk 5]
```diff
@@ -1,28 +1,46 @@
...
         id: validate
-        run: terraform validate -no-color
\ No newline at end of file
+        run: terraform validate -no-color
+
+      - name: Validate terraform fmt
+        run: |
+          set +e 
+          echo "CHECK_STATUS=0" >> "$GITHUB_ENV"
+          terraform fmt -recursive -check
```
[the remark this answers is not in the document]
Comment (gatici) [reply .github/workflows/terraform.yaml:38]: Ok, we can add diff.

File: .github/workflows/terraform.yaml [hunk 6]
```diff
@@ -1,28 +1,46 @@
...
         uses: hashicorp/setup-terraform@v3
 
-      - name: Terraform Format
-        id: fmt
-        run: terraform fmt -recursive -check
-        continue-on-error: true
-
-      - name: Terraform Init
+      - name: Initialize Terraform Module
         id: init
```
[the remark this answers is not in the document]
Comment (gatici) [reply .github/workflows/terraform.yaml:27]: I think this step helps to validate provider version.  Please check the output of workflows from a sample execution.
https://github.com/canonical/mongodb-operator/actions/runs/7888039487/job/21524616578

File: .github/workflows/terraform.yaml [hunk 7]
```diff
@@ -1,28 +1,46 @@
...
         run: terraform init
 
-      - name: Terraform Validate
+      - name: Validate Terraform Module
         id: validate
```
[the remark this answers is not in the document]
Comment (gatici) [reply .github/workflows/terraform.yaml:31]: Terraform file's syntax and structure are checked using validate.  If there is something wrong this step will fail.

File: .github/workflows/terraform.yaml (hunk 3 above)
Comment (gatici) [reply .github/workflows/terraform.yaml:46]: Removed this step and appended it to Validate terraform fmt.

File: .github/workflows/terraform.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (Gmerold) [reply .github/workflows/terraform.yaml:27]: No, I mean the `id` specifically. It's typically used to refer this step in another one. We don't do that.

File: .github/workflows/terraform.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (Gmerold) [reply .github/workflows/terraform.yaml:31]: As above.

File: .github/workflows/terraform.yaml (hunk 3 above)
Comment (Gmerold) [reply .github/workflows/terraform.yaml:46]: Did you push the changes? :)

File: .github/workflows/terraform.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (gatici) [reply .github/workflows/terraform.yaml:25]: OK

**Comment by gatici** [comment]
https://github.com/canonical/mongodb-operator/actions/runs/7902778399/job/21569170221 this is a sample execution which catches the formatting issues.

File: LICENSE [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2023 Canonical Ltd.
```
Comment (Gmerold) [inline LICENSE:190]:
```suggestion
   Copyright 2024 Canonical Ltd.
```

**Review by Gmerold — APPROVED** [review]

File: LICENSE (hunk 8 above)
Comment (gatici) [reply LICENSE]: Done

**Pull request merged**